lib/signs/utilities/early_am_suppression.ex

```diff
@@ -202,9 +202,19 @@ defmodule Signs.Utilities.EarlyAmSuppression do
                    scheduled_time: scheduled
                  }}
 
-              {headway_top, headway_bottom} ->
-                # Make sure routes is nil so that destination is used as headsign
-                {%{headway_top | destination: destination, routes: nil}, headway_bottom}
+              {headway_top, %Headways.Bottom{range: {_, high}} = headway_bottom} ->
+                # Check against schedule to account for mezzanine sources having different first scheduled arrivals
+                if DateTime.add(current_time, high, :minute) |> DateTime.compare(scheduled) == :gt do
```

Comment:
Just to check my understanding: This wouldn't be susceptible to the opposite problem, showing a timestamp when we should show headways, because the headways code takes the earliest time from all the stops for the sign. So it will always be available when we need it, but sometimes we need to suppress it again for 1/2 of the mezzanine sign. Right?

Reply:
Correct, headways will always be returned based on the earlier first scheduled arrival between the two source configs. We'll only sometimes have a need to suppress headways for the half associated with the later first scheduled arrival when that other half is less than 40 min from the first scheduled arrival, but more than [upper headway range] min from the first first scheduled arrival **and** there are no valid predictions to be shown.

This is probably a rather rare case on heavy rail since trains are staged long before the first scheduled arrivals resulting in terminal predictions. It's going to be more common on the GL, in particular GLX stations, since we don't have terminal predictions there yet.